app/models/recurring_todos/abstract_repeat_pattern.rb

```diff
@@ -174,7 +174,7 @@ def continues_recurring?(previous)
     # same day as the previous
     def determine_start(previous, offset=0.day)
       start = self.start_from || NullTime.new
-      now = Time.zone.now
+      now = Time.now
```

Comment:
I've found the problem! It's the application_controller that sets the TZ based on the logged in user_preference, and then is not reverted.

I've added the site timezone as the default in the test helper and I've added a line in the fixtures in which we set the timezone again as the site default, so that the comparisons make sense, what do you think?